cdap-credential-ext-gcp-wi/src/main/java/io/cdap/cdap/security/spi/credential/GcpWorkloadIdentityCredentialProvider.java

```diff
@@ -215,15 +214,16 @@ public ProvisionedCredential provision(NamespaceMeta namespaceMeta,
   }
 
   @VisibleForTesting
-  ProvisionedCredential getProvisionedCredential(NamespaceMeta namespaceMeta,
-      CredentialIdentity identity, @Nullable String scopes) throws IOException, ApiException {
+  ProvisionedCredential getProvisionedCredential(String namespace,
+      CredentialIdentity identity, CredentialProvisionContext context)
+      throws IOException, ApiException {
 
     // Get k8s namespace from namespace metadata if using a non-default namespace and namespace
     // creation hook is enabled.
     String k8sNamespace = NamespaceId.DEFAULT.getNamespace();
-    if (!namespaceMeta.getName().equals(NamespaceId.DEFAULT.getNamespace())
+    if (!namespace.equals(NamespaceId.DEFAULT.getNamespace())
```

Comment:
This comparison looks incorrect, `namespace` will always be `SYSTEM` here right?

We need to check the actual namespace from where the request was triggered in task worker.

Reply:
For example, in HDF where we don't set `k8s.namespace` property in `default` namespace but identities are present in `system` namespace, this condition will return true.

Reply:
Good catch. I addd a new field to encapsulate this value.